lib/electric_slide/call_queue.rb

```diff
@@ -151,7 +151,8 @@ def check_for_connections
     # @param [Adhearsion::Call] call Caller to be added to the queue
     def priority_enqueue(call)
       # Don't reset the enqueue time in case this is a re-insert on agent failure
-      call[:enqueue_time] ||= Time.now
+      call[:electric_slide_enqueued_at] ||= DateTime.now
+      call.on_end { abandon call }
```

Comment:
If we're going to use `#abandon` to mean any ending of the call, whether before connecting to an agent or after, we need to rename this method. For example, the log message inside `#abandon` logs the fact that it is an abandon. Abandon in this context specifically means "left the queue before getting to an agent."  This is an important fact to track, so we should probably update the method to check whether a connection occurred, and log an appropriate message.